### PR TITLE
added documentation about the format of certificateKey

### LINKED ISSUE #2928: add more documentation about the format of  certificateKey
SataQiu added validation for certificateKey:
https://github.com/kubernetes/kubernetes/blob/master/cmd/kubeadm/app/apis/kubeadm/validation/validation.go#L619-L635

but we should also add information about the format of the certificateKey in the API godocs and on the HA page at the website.

- [x] PR for kubernetes/kubernetes: add comment about the size and format over the certificateKey fields in v1beta3 and v1beta4 and flags: https://github.com/kubernetes/kubernetes/pull/120546

add new sentence:
```
The certificate key is a hex encoded string that is an AES key of size 32 bytes.
```

near the comment of `CertificateKey` instances in these files:
https://github.com/kubernetes/kubernetes/blob/master/cmd/kubeadm/app/apis/kubeadm/v1beta3/types.go
https://github.com/kubernetes/kubernetes/blob/master/cmd/kubeadm/app/apis/kubeadm/v1beta4/types.go
https://github.com/kubernetes/kubernetes/blob/master/cmd/kubeadm/app/apis/kubeadm/types.go

should be 2 per file, 6 total.

also add the same sentence in the --certificate-key flag description here:
https://github.com/kubernetes/kubernetes/blob/f8ca582a06a8379968ddb7960adefeea3c467e36/cmd/kubeadm/app/cmd/init.go#L191
https://github.com/kubernetes/kubernetes/blob/f8ca582a06a8379968ddb7960adefeea3c467e36/cmd/kubeadm/app/cmd/join.go#L254

- [x] PR for kuberntes/website: here:
https://kubernetes.io/docs/setup/production-environment/tools/kubeadm/high-availability/#stacked-control-plane-and-etcd-nodes https://github.com/kubernetes/website/pull/42981

under the note:
````
Note: The kubeadm-certs Secret and decryption key expire after two hours.
````

write a new note:
```
The certificate key is a hex encoded string that is an AES key of size 32 bytes.
````

### PR DESCRIPTION
**Pull request by SaumyaBhushan**
#### What type of PR is this?
/kind documentation

#### What this PR does / why we need it:
add more documentation about the format of certificateKey
#### Which issue(s) this PR fixes:

Fixes [#2928](https://github.com/kubernetes/kubeadm/issues/2928)

#### Does this PR introduce a user-facing change?

```release-note

```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by neolit123** [comment]
> Fixes #2928

please change to `xref https://github.com/kubernetes/kubeadm/issues/2928`

File: cmd/kubeadm/app/cmd/init.go [hunk 1]
```diff
@@ -188,7 +188,7 @@ func AddInitConfigFlags(flagSet *flag.FlagSet, cfg *kubeadmapiv1.InitConfigurati
 	)
 	flagSet.StringVar(
 		&cfg.CertificateKey, options.CertificateKey, "",
-		"Key used to encrypt the control-plane certificates in the kubeadm-certs Secret.",
+		"The certificate key is a hex encoded string that is an AES key of size 32 bytes. Key used to encrypt the control-plane certificates in the kubeadm-certs Secret.",
```
Comment (neolit123) [inline cmd/kubeadm/app/cmd/init.go:191]: the sentence must be after the existing sentence.

**Review by neolit123 — COMMENTED** [review]
please make all changes explained in https://github.com/kubernetes/kubeadm/issues/2928 for the kubernetes/kubernetes repository in a single commit in a single PR.

**Comment by SaumyaBhushan** [comment]
Hey @neolit123 Updated all the changes in single PR here https://github.com/kubernetes/kubernetes/pull/120546 .

**Comment by neolit123** [comment]
> Hey @neolit123 Updated all the changes in single PR here #120546 .

/close

**Pull request closed without merging**